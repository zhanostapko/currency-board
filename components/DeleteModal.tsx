"use client";
import React, { useEffect } from "react";
import Button from "./Shared/Button";

type DeleteModalProps = {
  currency: string;
  onConfirm: (currency: string) => void;
  onCancel: () => void;
};

const DeleteModal = ({ currency, onConfirm, onCancel }: DeleteModalProps) => {
  useEffect(() => {
    const handleKeyDown = (event: KeyboardEvent) => {
      if (event.key === "Escape") {
        onCancel();
      }
    };

    document.addEventListener("keydown", handleKeyDown);
    return () => {
      document.removeEventListener("keydown", handleKeyDown);
    };
  }, [onCancel]);

  return (
    <div className="fixed inset-0 flex items-center justify-center bg-black bg-opacity-50 p-6">
      <div className="max-w-lg w-full bg-white p-8 rounded-xl shadow-lg border border-gray-200 text-center">
        <h2 className="text-2xl font-bold text-gray-900 mb-4">
          Confirm Deletion
        </h2>
        <p className="text-gray-600 mb-6">
          Are you sure you want to delete <strong>{currency}</strong> from the
          list?
        </p>

        <div className="flex justify-center gap-4">
          <Button
            onClick={() => onConfirm(currency)}
            className="px-5 py-2.5 text-sm bg-red-500 text-white rounded-md hover:bg-red-600 transition"
          >
            Delete
          </Button>
          <Button
            onClick={onCancel}
            className="px-5 py-2.5 text-sm border border-gray-300 bg-white text-gray-700 rounded-md hover:bg-gray-100 transition"
          >
            Cancel
          </Button>
        </div>
      </div>
    </div>
  );
};

export default DeleteModal;
