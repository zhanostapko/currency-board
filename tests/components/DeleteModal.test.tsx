import { render, screen, fireEvent } from "@testing-library/react";
import DeleteModal from "../../components/DeleteModal";

describe("DeleteModal Component", () => {
  const mockOnConfirm = jest.fn();
  const mockOnCancel = jest.fn();
  const testCurrency = "USD";

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders modal with correct currency", () => {
    const { asFragment } = render(
      <DeleteModal
        currency={testCurrency}
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
      />
    );

    expect(screen.getByText("Confirm Deletion")).toBeInTheDocument();
    expect(
      screen.getByText(/Are you sure you want to delete/i)
    ).toBeInTheDocument();
    expect(screen.getByText(/USD/i)).toBeInTheDocument();
    expect(screen.getByText(/from the list/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("calls onConfirm when 'Delete' button is clicked", () => {
    render(
      <DeleteModal
        currency={testCurrency}
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
      />
    );

    fireEvent.click(screen.getByText("Delete"));

    expect(mockOnConfirm).toHaveBeenCalledTimes(1);
    expect(mockOnConfirm).toHaveBeenCalledWith(testCurrency);
  });

  test("calls onCancel when 'Cancel' button is clicked", () => {
    render(
      <DeleteModal
        currency={testCurrency}
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
      />
    );

    fireEvent.click(screen.getByText("Cancel"));

    expect(mockOnCancel).toHaveBeenCalledTimes(1);
  });

  test("calls onCancel when Escape key is pressed", () => {
    render(
      <DeleteModal
        currency={testCurrency}
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
      />
    );

    fireEvent.keyDown(document, { key: "Escape", code: "Escape" });

    expect(mockOnCancel).toHaveBeenCalledTimes(1);
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <DeleteModal
        currency={testCurrency}
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
