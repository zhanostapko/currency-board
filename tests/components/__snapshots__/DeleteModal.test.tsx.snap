// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 flex items-center justify-center bg-black bg-opacity-50 p-6"
  >
    <div
      class="max-w-lg w-full bg-white p-8 rounded-xl shadow-lg border border-gray-200 text-center"
    >
      <h2
        class="text-2xl font-bold text-gray-900 mb-4"
      >
        Confirm Deletion
      </h2>
      <p
        class="text-gray-600 mb-6"
      >
        Are you sure you want to delete 
        <strong>
          USD
        </strong>
         from the list?
      </p>
      <div
        class="flex justify-center gap-4"
      >
        <button
          class="px-5 py-2.5 text-sm bg-red-500 text-white rounded-md hover:bg-red-600 transition"
        >
          Delete
        </button>
        <button
          class="px-5 py-2.5 text-sm border border-gray-300 bg-white text-gray-700 rounded-md hover:bg-gray-100 transition"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DeleteModal Component renders modal with correct currency 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 flex items-center justify-center bg-black bg-opacity-50 p-6"
  >
    <div
      class="max-w-lg w-full bg-white p-8 rounded-xl shadow-lg border border-gray-200 text-center"
    >
      <h2
        class="text-2xl font-bold text-gray-900 mb-4"
      >
        Confirm Deletion
      </h2>
      <p
        class="text-gray-600 mb-6"
      >
        Are you sure you want to delete 
        <strong>
          USD
        </strong>
         from the list?
      </p>
      <div
        class="flex justify-center gap-4"
      >
        <button
          class="px-5 py-2.5 text-sm bg-red-500 text-white rounded-md hover:bg-red-600 transition"
        >
          Delete
        </button>
        <button
          class="px-5 py-2.5 text-sm border border-gray-300 bg-white text-gray-700 rounded-md hover:bg-gray-100 transition"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
